applications/fishing-map/src/features/analysis/Analysis.tsx

```diff
@@ -0,0 +1,243 @@
+  CreateReport,
+  createReportThunk,
+  DateRange,
+  ReportGeometry,
+  resetReportStatus,
+  selectAnalysisGeometry,
+  selectReportStatus,
+  setAnalysisGeometry,
+} from './analysis.slice'
+import AnalysisGraphWrapper from './AnalysisGraphWrapper'
+
+function Analysis() {
+  const { t } = useTranslation()
+  const workspaceStatus = useSelector(selectWorkspaceStatus)
+  const dispatch = useDispatch()
+  const mapInstance = useMapboxInstance()
+  const fitMapBounds = useMapFitBounds()
+  const { setMapCoordinates } = useViewport()
+  const tilesLoading = useTilesState(mapInstance as Map).loading
+  const debouncedTilesLoading = useDebounce(tilesLoading, 600)
+  const analysisQuery = useSelector(selectAnalysisQuery)
+  const [sourceLoaded, setSourceLoaded] = useState(false)
+  const timeoutRef = useRef<NodeJS.Timeout>()
+  const { dispatchQueryParams } = useLocationConnect()
+  const { updateFeatureState, cleanFeatureState } = useFeatureState(mapInstance)
+  const staticTime = useSelector(selectStaticTime)
+  const dataviews = useSelector(selectTemporalgridDataviews) || []
+  const analysisGeometry = useSelector(selectAnalysisGeometry)
+  const reportStatus = useSelector(selectReportStatus)
+  const userData = useSelector(selectUserData)
+  const hasAnalysisLayers = useSelector(selectHasAnalysisLayersVisible)
+  const { areaId, sourceId } = analysisQuery
+
+  useLayoutEffect(() => {
+    // We have to go to zoom 0 to load every geometry to be able to fitBounds later
+    const viewport = { latitude: 0, longitude: 0, zoom: 0 }
+    setMapCoordinates(viewport)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    const sourceCallback = () => {
+      if (mapInstance?.getSource(sourceId) && mapInstance?.isSourceLoaded(sourceId)) {
+        setSourceLoaded(true)
+        mapInstance.off('idle', sourceCallback)
+      }
+    }
+
+    if (mapInstance && sourceId && areaId) {
+      setSourceLoaded(false)
+      mapInstance.on('idle', sourceCallback)
+    }
+
+    return () => {
+      if (mapInstance) {
+        mapInstance.off('idle', sourceCallback)
+      }
+    }
+  }, [mapInstance, sourceId, areaId])
+
+  useEffect(() => {
+    if (sourceLoaded && mapInstance) {
+      const contextAreaFeatures = mapInstance?.querySourceFeatures(sourceId, {
+        sourceLayer: 'main', // Our layers always uses 'main' as the source layer
+        filter: ['==', 'gfw_id', parseInt(areaId)],
+      })
+      if (contextAreaFeatures.length > 0) {
+        cleanFeatureState('highlight')
+        const contextAreaFeatureMerged = contextAreaFeatures.reduce(
+          (acc, { geometry, properties }) => {
+            const featureGeometry: Feature<Polygon> = {
+              type: 'Feature',
+              geometry: geometry as Polygon,
+              properties,
+            }
+            if (!acc?.type) return featureGeometry
+            return union(acc, featureGeometry, { properties }) as Feature<Polygon>
```

Comment:
Is possible that there is an error in this line?

> Argument of type '{ properties: GeoJsonProperties; }' is not assignable to parameter of type 'Feature<Polygon, Properties>'.
>   Type '{ properties: GeoJsonProperties; }' is missing the following properties from type 'Feature<Polygon, Properties>': type, geometryts(2345)
>

Reply:
Besides that, I wonder how would the `union` method merge the properties from multiple polygons. I think it will just leave the `properties` from the last item in the `contextAreaFeatures` array, then it'll use that for the name of the AnalysisGeometry. Is that ok?

Reply:
> Is possible that there is an error in this line?

I haven't had that error, did you run `yarn` in the project to ensure all dependencies were updated?

> I think it will just leave the properties from the last item in the contextAreaFeatures array

you are right, I'm just doing it to ensure we don't miss any part of the geometry because the documentation says:

"The domain of the query includes all currently-loaded vector tiles and GeoJSON source tiles: this function does not check tiles outside the currently visible viewport.
Because features come from tiled vector data or GeoJSON data that is converted to tiles internally, feature geometries may be split or duplicated across tile boundaries and, as a result, features may appear multiple times in query results."

but properties should be the same in all the features so we are safe to use it for the name